src/components/tasks/crud/createTask.js

```diff
@@ -57,60 +58,69 @@ const basicSwal = {
 
 const MySwal = withReactContent(Swal)
 
-export const createTask = (tasks, reload, setReload) => {
+export const createTask = (onCreateTask, setReload) => {
     MySwal.fire({
         ...basicSwal,
         html: <AddDescription
             handleChange={{'md': handleMarkdownChange, 'text': handleTextChange, 'file': handleFileChangeDescription}}/>
     }).then(result => {
         if (result.isConfirmed) {
-            createTests(tasks, reload, setReload)
+            createTests(onCreateTask, setReload)
         }
     })
 }
 
 
 const handleManualTestsChange = (testCases) => {
     delete task['testsBase64']
+    task['testsBase64'] = btoa(JSON.stringify(testCases))
 }
 
-const createTests = (tasks, reload, setReload) => {
+const createTests = (onCreateTask, setReload) => {
     MySwal.fire({
         ...basicSwal,
         html: <AddTests handleChange={{'file': handleFileChangeTests, 'manual': handleManualTestsChange}}/>
     }).then(result => {
         if(result.isConfirmed) {
-            createTimeLimit(tasks, reload, setReload)
+            createTimeLimit(onCreateTask, setReload)
         }
     })
 }
 
-const createTimeLimit = (tasks, reload, setReload) => {
+const createTimeLimit = (onCreateTask, setReload) => {
     MySwal.fire({
         ...basicSwal,
         input: 'number',
         text: 'Add time limit'
     }).then(result => {
         if(result.isConfirmed) {
             task['timeLimit'] = Number(result.value)
-            createDeadline(tasks, reload, setReload)
+            createDeadline(onCreateTask, setReload)
         }
     })
 }
 
-const createDeadline = (tasks, reload, setReload) => {
+const createDeadline = (onCreateTask, setReload) => {
     MySwal.fire({
         ...basicSwal,
         html: <AddDeadline handleChange={(date) => task['deadline'] = date} />,
         confirmButtonText: 'Create'
     }).then(result => {
         if(result.isConfirmed) {
-            create(tasks, reload, setReload)
+            create(onCreateTask, setReload)
         }
     })
 }
 
-const create = (tasks, reload, setReload) => {
-    tasksApi.updateTasks([...tasks, task])
-        .then(_ => setReload(!reload))
+const create = (onCreateTask, setReload) => {
+    withSwal({
+        promise: () => onCreateTask(task).then(resp => {
+            setReload(reload => !reload)
+            return resp
+        }),
+        successSwalTitle: "Success",
+        successSwalText: "Successfully added new task",
+        errorSwalTitle: "Error",
```

Comment:
Ogólnie możesz zostawić same tytuły. Jeśli wejdzie `Success`, to będzie wiedział, czego to jest success, jeśli wejdzie error, to i tak nic nie wnosisz tą wiadomością. Jest zasada, że powinieneś użyć jak najmniej słów.